### ISSUE TITLE
More than two layers in the layerlist should be visible without scrolling on small screens

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#1710]
## Motivation

On my smaller laptop screen, I can just see two of the layer names. This is very cramped, especially if you're working with a reasonable set of layers. I would prefer to be able to see maybe three or four layers before having to scroll up or down. 

It's a bit frustrating that I can't resize the amount of relative space taken up by the layerlist vs the layer controls in the left hand menu bar, especially since there's often a blank bit of UI real estate at the bottom of the layer controls.

## Pitch

We should dedicate a bit more space to the layerlist in the left hand menu bar, relative to the layer controls above it.

### Workaround 1
The workaround I could use here is to "pop out" the layerlist, and then switch back and forth between tabs on the left hand menubar (eg: select a layer, switch tabs to change contrast or add points, etc.). This toggling is also kind of annoying to do.

### Workaround 2
If I move napari over to the larger second monitor then I can expand the size of the whole GUI & see more of the layers in the left hand menu bar. The reason I don't do that much is because there's some kind of glitch with the screen resolution or something on the second monitor - when I try to click on buttons I find I need to put the mouse cursor much higher on the screen for the click to register on the actual button. (I haven't put in a bug report about that problem, mostly because I'm not sure how to make a minimum reproducible example.)

## Additional context

Here are some screenshots to illustrate.

Screenshot showing only two layers are visible without scrolling in the list, while at the same time there is a lot of blank space in the image layer controls above.
<img width="959" alt="napari-screenshot-1" src="https://user-images.githubusercontent.com/30920819/95421873-02150180-098a-11eb-97f8-102a740ac25f.PNG">

I can swap the positions of the two GUI elements, but it doesn't help me change the relative size between them.
<img width="959" alt="napari-screenshot-3" src="https://user-images.githubusercontent.com/30920819/95421903-0c370000-098a-11eb-96ee-7ec8365a39a0.PNG">

I can "pop out" the layerlist, but then I need to tab back and forth to access the layer controls every time I select a layer, which is inconvenient.
<img width="958" alt="napari-screenshot-2" src="https://user-images.githubusercontent.com/30920819/95421889-06d9b580-098a-11eb-850d-2ff41381407d.PNG">

**Comment by Czaki** [comment]
Layer list could be moved with drag'n'drop. Moving to right is not a solution?

**Comment by GenevieveBuckley** [comment]
Well it's only a problem on a small screen, so moving it to the right just cuts into the available screen space even more. But sure, you can do that.

**Comment by GenevieveBuckley** [comment]
@DragaDoncila and I have worked out that changing the display scaling has a big effect on how cramped the napari layerlist in the GUI feels.

On my local machine, the settings are:
* Windows 10
* Display scale: 150% (Recommended)
* Display resolution: 1920 x 1080 (Recommended)
* Display orientation: Landscape

<img width="290" alt="DisplaySettins" src="https://user-images.githubusercontent.com/30920819/96400633-fc86a980-121c-11eb-8a8b-1727d7439f41.PNG">

Here's what happens when I change the display scaling. At 100% scaling, there's lots of room to see multiple layers in the layerlist:
![napari-screenshot-100pct-scaling](https://user-images.githubusercontent.com/30920819/96400684-1de79580-121d-11eb-89ae-04f9a6268fe1.png)

But when the scaling is 150% (the recommended setting for this machine), things are much more cramped:
![napari-screenshot-150pct-scaling](https://user-images.githubusercontent.com/30920819/96400737-41124500-121d-11eb-8d3e-69e329a76215.png)

**Comment by jni** [comment]
@GenevieveBuckley thanks for those screenshots, they are very useful! That looks like >150% change in size to me, which is weird! At any rate, I think we should allow UI element rescaling independent (or on top of) the OS setting.

**Comment by Czaki** [comment]
It looks like double scaling. (rather 2.25 than 1.5)

**Comment by Czaki** [comment]
@GenevieveBuckley Could you comment out this line 
https://github.com/napari/napari/blob/0055113d7ffc68e06c032f9cdf22ed2fbf9db0c6/napari/_qt/event_loop.py#L58
and make screenshot?

**Comment by GenevieveBuckley** [comment]
> 
> 
> @GenevieveBuckley Could you comment out this line
> 
> https://github.com/napari/napari/blob/0055113d7ffc68e06c032f9cdf22ed2fbf9db0c6/napari/_qt/event_loop.py#L58
> 
> and make screenshot?

I tried this, and the screenshot results were the same. No difference there!

**Comment by GenevieveBuckley** [comment]
> 
> 
> @GenevieveBuckley thanks for those screenshots, they are very useful! That looks like >150% change in size to me, which is weird!

Yeah, the other thing I notice is that the change in size isn't linear with the amount of scaling. There seems to be two modes: "tiny" and "gigantic". With scaling from 100% - 125% everything in napari is pretty tiny, but everything with scaling from 150% - 175% looks gigantic.

The options for display scaling are: 100%, 125%, 150%, and 175%. Here's what that looks like on my screen:

### Display scaling 100%
![napari-screenshot-100pct-scaling](https://user-images.githubusercontent.com/30920819/96520982-f72c6c00-12bb-11eb-9a3f-4339c6eb4f24.png)

### Display scaling 125%
![napari-screenshot-125pct-scaling](https://user-images.githubusercontent.com/30920819/96521006-027f9780-12bc-11eb-9499-74144f6085d3.png)

### Display scaling 150%
![napari-screenshot-150pct-scaling](https://user-images.githubusercontent.com/30920819/96521030-0c08ff80-12bc-11eb-9c16-c11ff40fbdcb.png)

### Display scaling 175%
![napari-screenshot-175pct-scaling](https://user-images.githubusercontent.com/30920819/96521045-14613a80-12bc-11eb-823f-0a681c19338f.png)

**Comment by DragaDoncila** [comment]
@GenevieveBuckley does going down to 100% or 125% where the display isn't cramped affect your issue with the mouse cursor being "off"?

**Comment by Czaki** [comment]
Did you try to create some simple QT application (window with few buttons) and check ist behaviour? To check if it is Qt or napari specific bug?

**Comment by GenevieveBuckley** [comment]
> @GenevieveBuckley does going down to 100% or 125% where the display isn't cramped affect your issue with the mouse cursor being "off"?

I have the mouse cursor issue on the other monitor, not on this one.

**Comment by GenevieveBuckley** [comment]
> 
> 
> Did you try to create some simple QT application (window with few buttons) and check ist behaviour? To check if it is Qt or napari specific bug?

No, I haven't. The issue I'm raising here is that the layerlist is given proportionally far less space in the napari viewer when the display scaling is ~150%, and the layerlist is a specific napari thing.

If you want to know if the scaling of all Qt apps isn't linear with the display scaling, I don't know. If you think that would help for debugging and have a hello world type app, I could try it and take some screenshots for you.

**Comment by Czaki** [comment]
I try to run this and on 150% scale I got:
```
C:\Users\Grzesiek\PycharmProjects\napari\venv\Scripts\python.exe -m napari

Process finished with exit code -1073741819 (0xC0000005)
```

**Comment by Czaki** [comment]
@GenevieveBuckley Did you use PySide2 or PyQt5? If PySide2 could You try PyQt5? I got segfault on PySide2, but it look quit good on PyQt5 and I see difference between 100% and 125% scaling factor. 

And as I see the problem is that layer property dock widget has fixed size (without support of scroll of its content).

**Comment by GenevieveBuckley** [comment]
> 
> 
> @GenevieveBuckley Did you use PySide2 or PyQt5? If PySide2 could You try PyQt5? I got segfault on PySide2, but it look quit good on PyQt5 and I see difference between 100% and 125% scaling factor.

I've tried this with PyQt5 and PySide2 and get the same behaviour. The screenshots above are a pretty accurate representation of what happens on my machine.

> And as I see the problem is that layer property dock widget has fixed size (without support of scroll of its content).

You might be onto something here. Seems likely this might be the culprit!

**Comment by Czaki** [comment]
> I've tried this with PyQt5 and PySide2 and get the same behaviour. The screenshots above are a pretty accurate representation of what happens on my machine.

on Qt 5.15.1 I have different behavior for PyQt5 and PySide2

**Comment by GenevieveBuckley** [comment]
I thought you said you had a segfault with PySide2 (i.e. PySide2 wasn't working for you at all). Can you describe the difference in behaviour that you see?

**Comment by Czaki** [comment]
When I use PyQt 5.15.1 and select scaling 125% size of components change. On PySide2 there is no change of sizes. 
For PySide2 5.15.1 I got segfault for 150%. for PySide2 5.14.2 I got proper scaling.

**Comment by brisvag** [comment]
Now that #4368 is merged, this should be easier to solve by simply lowering the minimum size for the layer controls widget. Not sure if there is design discussion to be had around this though :)
